core/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -618,16 +618,16 @@ private static void parseNullValue(ParseContext context, ObjectMapper parentMapp
 
     private static Mapper.Builder<?,?> createBuilderFromFieldType(final ParseContext context, MappedFieldType fieldType, String currentFieldName) {
         Mapper.Builder builder = null;
-        if (fieldType instanceof StringFieldType) {
```

Comment:
This change also makes sense to me, but I'd like @jpountz to take a look at this because I wonder what we might have missed here in terms of tests and also to determine if we should backport this change to other branches. It looks like we never reached the TextFieldType-case in the past and I'm wondering what that means of if it is harmless on the long run.

Reply:
Sure, I have amended my commit here. I will open another PR with just this change.

Reply:
Done at https://github.com/elastic/elasticsearch/pull/27726